adapters/http-vertx-base/src/test/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapterTest.java

```diff
@@ -527,6 +527,46 @@ public void testUploadCommandResponseWaitsForAcceptedOutcome() {
                 any());
     }
 
+    /**
+     * Verifies that the adapter accepts a command response message with an empty body.
+     */
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testUploadEmptyCommandResponseSucceeds() {
+
+        // GIVEN an adapter with a downstream application attached
+        final CommandResponseSender sender = mock(CommandResponseSender.class);
+        when(sender.sendCommandResponse(any(CommandResponse.class), (SpanContext) any())).thenReturn(Future.succeededFuture(mock(ProtonDelivery.class)));
+
+        when(commandConsumerFactory.getCommandResponseSender(anyString(), anyString())).thenReturn(Future.succeededFuture(sender));
+
+        final HttpServer server = getHttpServer(false);
+        final AbstractVertxBasedHttpProtocolAdapter<HttpProtocolAdapterProperties> adapter = getAdapter(server, null);
+
+        // WHEN a device publishes a command response with an empty payload
+        final Buffer payload = Buffer.buffer();
```

Comment:
I am not sure if this is the same as *no payload*. Can you change this to `null`?

Reply:
From what I understand Netty (and thus Vert.x MQTT) does not support this: 
https://github.com/netty/netty/blob/56055f4404fbebeb6403165d352d62330a1bf81d/codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttPublishMessage.java#L41-L52
The [MQTT 3.1.1 standard](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718040) says: " It is valid for a PUBLISH Packet to contain a zero length payload." Which of course does not tell how this is implemented... 
Hono's MQTT adapter will throw null pointer exceptions.

Reply:
@sophokles73 If you prefer I can still change the test so that the payload is `null` and fix the (few) places where this would cause NPEs in the HTTP and MQTT adapters.

Reply:
no